frontend/apps/reader/modules/readerwikipedia.lua

```diff
@@ -172,79 +172,124 @@ function ReaderWikipedia:addToMainMenu(menu_items)
-                                            })
+                                        if not has_subdirectory then
+                                            dir = dir:match("(.*)/")
                                         end
-                                    }
-                                    UIManager:show(path_chooser)
-                                end,
+                                        local PathChooser = require("ui/widget/pathchooser")
+                                        local path_chooser = PathChooser:new{
+                                            title = _("Wikipedia 'Save as EPUB' directory"),
+                                            path = dir,
+                                            show_hidden = G_reader_settings:readSetting("show_hidden"),
+                                            onConfirm = function(path)
+                                                -- hack to remove additional parent
+                                                if path:sub(-3, -1) == "/.." then
+                                                    path = path:sub(1, -4)
+                                                end
+                                                path = require("ffi/util").realpath(path)
+                                                G_reader_settings:saveSetting("wikipedia_save_dir", path)
+                                                UIManager:show(InfoMessage:new{
+                                                    text = T(_("Wikipedia 'Save as EPUB' directory set to:\n%1"), path),
+                                                })
+                                            end
+                                        }
+                                        UIManager:show(path_chooser)
+                                    end,
+                                    },
                                 },
                             },
-                        },
-                    }
-                    UIManager:show(dialog)
+                        }
+                        UIManager:show(dialog)
+                    end
+                    -- If wikipedia_save_dir has not yet been set, propose to use
+                    -- home_dir/Wikipedia/
+                    if not G_reader_settings:readSetting("wikipedia_save_dir") then
+                        local home_dir = G_reader_settings:readSetting("home_dir")
+                        if not home_dir or not lfs.attributes(home_dir, "mode") == "directory" then
+                            home_dir = require("apps/filemanager/filemanagerutil").getDefaultDir()
+                        end
+                        home_dir = home_dir:gsub("^(.-)/*$", "%1") -- remove trailing slash
+                        if home_dir and lfs.attributes(home_dir, "mode") == "directory" then
+                            local wikipedia_dir = home_dir.."/Wikipedia"
+                            local text = _([[
+
```

Comment:
I assume the extra newline serves some purpose?

Reply:
If we want more padding it should be part of the widget, not hacked with newlines imo. Also, my bad. :-P

Reply:
I agree.
And the air is not even that needed, and no real need to add these to widgets.
The visual feeling depends on the length of the lines, as they are not justified.
When you have short lines, and so some free space on the right, it feels prettier when you add some space at top and bottom (the left being used by the image).
But you can't really know how short your lines will be on the different devices, so let's do as we did with no newlines and no additional padding.